connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaConnectorProperties.java

```diff
@@ -8,34 +8,87 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 import io.camunda.connector.feel.annotation.FEEL;
+import io.camunda.connector.generator.dsl.Property;
+import io.camunda.connector.generator.java.annotation.TemplateProperty;
 import io.camunda.connector.kafka.model.Avro;
 import io.camunda.connector.kafka.model.KafkaAuthentication;
 import io.camunda.connector.kafka.model.KafkaTopic;
+import io.camunda.connector.kafka.model.SerializationType;
 import jakarta.validation.Valid;
 import jakarta.validation.constraints.NotNull;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public final class KafkaConnectorProperties {
-
-  @NotNull private AuthenticationType authenticationType;
-
-  @Valid private KafkaAuthentication authentication;
-
-  @Valid @NotNull private KafkaTopic topic;
-
-  @FEEL private Map<String, Object> additionalProperties = new HashMap<>();
-
-  private String activationCondition;
-
-  @FEEL private List<Long> offsets;
-
-  @NotNull private AutoOffsetReset autoOffsetReset = AutoOffsetReset.NONE;
-
-  private String groupId;
-
-  @Valid private Avro avro;
+public record KafkaConnectorProperties(
+    @TemplateProperty(
+            group = "kafka",
+            label = "Deserialization type",
+            id = "serializationType",
+            defaultValue = "json",
+            type = TemplateProperty.PropertyType.Dropdown,
+            choices = {
+              @TemplateProperty.DropdownPropertyChoice(value = "json", label = "Default (JSON)"),
+              @TemplateProperty.DropdownPropertyChoice(
+                  value = "avro",
+                  label = "AVRO (experimental)")
```

Comment:
Avro instead of AVRO? 
Is this documented? Even if it's experimental we might want to guide the users, but maybe it's already documented

Reply:
Nice! thank you 🙏